Doppler.ContactPolicies.Business.Logic/Services/ContactPoliciesService.cs

```diff
@@ -21,5 +22,36 @@ public async Task<ContactPoliciesSettingsDto> GetContactPoliciesSettingsAsync(st
                 (await _contactPoliciesSettingsRepository.GetContactPoliciesSettingsAsync(accountName)).ToDto();
             return contactPoliciesSettings;
         }
+
+        public async Task<bool> InsertContactPoliciesSettingsAsync(string accountName, ContactPoliciesSettingsDto contactPoliciesSettings)
+        {
+            var existingUsercontactPolicies = await _contactPoliciesSettingsRepository.GetBasicContactPoliciesSettingsAsync(accountName);
+
+            /// User doesnt exist
+            if (existingUsercontactPolicies is null)
+                return false;
+
+            /// User doesnt have permissions
+            if (existingUsercontactPolicies.IdUser is null)
+                throw new Exception($"This action is not allowed for the user with Account {accountName}.");
+
+            var contactPoliciesToInsert = CreateUserContactPoliciesToInsert(contactPoliciesSettings, existingUsercontactPolicies);
+            var isSuccsesfulyInserted = await _contactPoliciesSettingsRepository.InsertContactPoliciesSettingsAsync(contactPoliciesToInsert);
+
+            return isSuccsesfulyInserted;
+        }
+
+        private ContactPoliciesSettings CreateUserContactPoliciesToInsert(ContactPoliciesSettingsDto contactPoliciesSettings, ContactPoliciesSettings existingUsercontactPolicies)
+        {
+            return new ContactPoliciesSettings
+            {
+                AccountName = existingUsercontactPolicies.AccountName,
+                IdUser = existingUsercontactPolicies.IdUser,
```

Comment:
Oh, [I understand it now](https://github.com/FromDoppler/doppler-contact-policies-api/pull/21/files#r659725907).

What do you think about simply passing the account name and id user in place of a `ContactPoliciesSettings`?

What is more, do you really need both identifiers? in my opinion, `AccountName` only should be enough. Am I missing something?

Reply:
It is also discussed in https://github.com/FromDoppler/doppler-contact-policies-api/pull/21#discussion_r660901804

Marking this conversation as resolved...